### PR TITLE
Recode Updates

### PR DESCRIPTION
**Pull request by TreemanKing**
- Fix error in folder creation.
- Use output file directly instead of asserting a new one

### CODE REVIEW

**Comment by TreemanKing** [comment]
- [ ] Fix injection in AdvancedPortal in core as it both tagRegistry and playerDataServices is null when attempted to be used

**Comment by TreemanKing** [comment]
- [] Bug: Currently cannot go through portals after creation due to throwback

**Comment by TreemanKing** [comment]
- [ ] `/advancedportals show` spams a lot of errors

**Review by sekwah41 — APPROVED** [review]

File: core/src/main/java/com/sekwah/advancedportals/core/tags/activation/NameTag.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import com.sekwah.advancedportals.core.warphandler.ActivationData;
 import com.sekwah.advancedportals.core.warphandler.Tag;
 
+import javax.annotation.Nullable;
 import java.util.List;
```
Comment (sekwah41) [inline core/src/main/java/com/sekwah/advancedportals/core/tags/activation/NameTag.java:10]: There was some super old issue with nullable which I just avoided but I believe it was the Jetbrains nullable xD I need to start properly using it.

File: spigot/src/main/java/com/sekwah/advancedportals/spigot/connector/container/SpigotServerContainer.java [hunk 2]
```diff
@@ -61,4 +62,12 @@ public PlayerContainer[] getPlayers() {
                 .map(SpigotPlayerContainer::new)
                 .toArray(PlayerContainer[]::new);
     }
+
+    // Check if it's a material compatible with making portals
+    private boolean isAdvancedPortalBlock(Material material) {
+        return switch (material) {
+            case WATER, LAVA, AIR, NETHER_PORTAL, END_GATEWAY, END_PORTAL -> true;
```
Comment (sekwah41) [inline spigot/src/main/java/com/sekwah/advancedportals/spigot/connector/container/SpigotServerContainer.java:69]: non full blocks also work, though these are the main ones. E.g. you can use iron bars or even stairs.

Though might be good to just include the main ones for now.

**Review by sekwah41 — APPROVED** [review]

**Pull request merged**